File: services/notification-service/src/kafka/event/event-type.ts
export enum EventType {
  // payment
  PAYMENT_RECEIVED = 'payment.received',
  INVOICE_CREATED = 'invoice.created',

  // ride
  RIDE_CREATED = 'ride.created',
  RIDE_COMPLETED = 'ride.completed',
  RIDE_STARTED = 'ride.started',
  RIDE_CANCELED = 'ride.canceled',

  // vehicle
  VEHICLE_CREATED = 'vehicle.created',
  VEHICLE_UPDATED = 'vehicle.updated',
  VEHICLE_ASSIGNED = 'vehicle.assigned',
  VEHICLE_ARRIVED = 'vehicle.arrived',
  VEHICLE_UNAVAILABLE = 'vehicle.unavailable',

  // user
  USER_UPDATED = 'user.updated',
  USER_CREATED = 'user.created',
}

File: services/vehicle-service/Events/EventType.cs
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum EventType
{
    [EnumMember(Value = "vehicle.created")]
    VehicleCreated,

    [EnumMember(Value = "vehicle.updated")]
    VehicleUpdated,

    [EnumMember(Value = "vehicle_activity.created")]
    VehicleActivityCreated,

    [EnumMember(Value = "vehicle_activity.updated")]
    VehicleActivityUpdated,

    [EnumMember(Value = "vehicle.assigned")]
    VehicleAssigned,

    [EnumMember(Value = "vehicle.arrived")]
    VehicleArrived,
    [EnumMember(Value = "vehicle.unavalailable")]
    VehicleUnavailable,
}
